engine/src/main/java/org/terasology/engine/TerasologyEngine.java

```diff
@@ -428,34 +429,34 @@ private void mainLoop() {
 
             Iterator<Float> updateCycles = time.tick();
 
-            PerformanceMonitor.startActivity("Network Update");
-            networkSystem.update();
-            PerformanceMonitor.endActivity();
+            try (Activity ignored = PerformanceMonitor.startActivity("Network Update")) {
```

Comment:
Thank you, good to know. Too bad about it, the syntax as it is is quite distracting from my perspective. But it will indeed make things easier when multiple return locations are involved and will make maintenance easier overall: the IDE will complain about missing matching brackets, while it won't normally complain if the endActivity() line is deleted for whatever reason.